app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt

```diff
@@ -152,11 +172,51 @@ class LaunchActivity : AppCompatActivity() {
         }
     }
 
-    private fun handleDeeplink() {
+    private fun handleDeeplink() = lifecycleScope.launch(Dispatchers.IO) {
         intent.data?.let { uri -> uri.path?.let { path -> processDeepLink(path) } }
+        canFinishActivity.postValue(true)
     }
 
     private fun processDeepLink(path: String) {
+        var trackerValue = ""
+        if (path.contains("/app/share/")) {
+            Regex("/app/share/(\\d+)/([a-z0-9-]+)").find(path)?.let { match ->
+                val (driveId, fileSharedLinkUuid) = match.destructured
+
+                val apiResponse = ApiRepository.getShareLinkInfo(driveId.toInt(), fileSharedLinkUuid)
+                when (apiResponse.result) {
+                    ApiResponseStatus.SUCCESS -> {
+                        val shareLink = apiResponse.data!!
+                        if (apiResponse.data?.validUntil?.before(Date()) == true) {
+                            Log.e("TOTO", "downloadSharedFile: expired | ${apiResponse.data?.validUntil}")
```

Comment:
I kept it to avoid having to constently stash and reapply this on each tests, it will be removed in the PR that take care of expired links
This PR will be merged in a feature branch, not on master so it's not a problem